crates/pallet-domains/src/lib.rs

```diff
@@ -1931,13 +1944,54 @@ mod pallet {
                 let consensus_transaction_byte_fee =
                     sp_domains::DOMAIN_STORAGE_FEE_MULTIPLIER * transaction_byte_fee;
 
-                let invalid_inherent_extrinsic_data = InvalidInherentExtrinsicData {
-                    extrinsics_shuffling_seed,
-                    timestamp,
-                    consensus_transaction_byte_fee,
-                };
+                // Record a storage proof for each domain with a successful bundle
+                for domain_id in SuccessfulBundles::<T>::iter_keys() {
+                    let domain_chain_allowlist =
+                        T::DomainAllowlistUpdates::domain_allowlist_updates(domain_id)
+                            .unwrap_or_default();
```

Comment:
Is it because of the allowlist update we need to store the `InvalidInherentExtrinsicData` for each domain separately?

While most of the data are the same for all the domain I think it makes more sense to store them once and shared by all the domains, for the allowlist update we can store `Map<domain_id, AllowlistUpdate>` which may bring redundant data in the FP but it should be fine as FP and the allowlist update are rare.